mesonbuild/utils/vsenv.py

```diff
@@ -66,7 +66,9 @@ def _setup_vsenv(force: bool) -> bool:
             '-utf8',
             '-format',
             'json'
-        ]
+        ],
+        encoding='utf8',
+        errors='replace',
```

Comment:
What is causing these errors and why is replacing them the best thing to do here?

Reply:
Actually there doesn't seem to be anything wrong here at all? The error came from the bat file, not vswhere. As far as vswhere is concerned, it has been told to emit utf-8, we read it as bytes, and json.loads() should always accept bytes of utf-8.

Reply:
Yes, I realized after that the error was coming from the second `check_output` below. But it is always a good habit to specify encoding, as default encoding is system dependent.

`replace` is arbitrary. It allows to see where the unhandled characters are. But could also be `ignore`, or `backslashreplace`. The goal is to prevent error if invalid characters are encountered.

Reply:
But we *do* specify encoding, we specify it via a `-utf8` flag to vswhere. And json.loads() explicitly defines the permitted encoding:
> Changed in version 3.6: s can now be of type [bytes](https://docs.python.org/3/library/stdtypes.html#bytes) or [bytearray](https://docs.python.org/3/library/stdtypes.html#bytearray). The input encoding should be UTF-8, UTF-16 or UTF-32.

(In the absence of a BOM it uses utf-8 to decode that bytestring.)

You're not "specifying the encoding", you are moving the location of the decode from the json module to here while trying to handle errors. The question is why do we think errors can ever happen -- again, vswhere -utf8 is supposed to guarantee the encoding is utf8 as expected by json.

Reply:
ok. reverted.